src/core/net/srp_server.cpp

```diff
@@ -1482,6 +1486,30 @@ const char *Server::AddressModeToString(AddressMode aMode)
     return kAddressModeStrings[aMode];
 }
 
+void Server::UpdateResponseCounters(Dns::UpdateHeader::Response aResponseCode)
+{
+    switch (aResponseCode)
+    {
+    case Dns::UpdateHeader::kResponseSuccess:
+        ++mResponseCounters.mSuccess;
+        break;
+    case Dns::UpdateHeader::kResponseServerFailure:
+        ++mResponseCounters.mServerFailure;
+        break;
+    case Dns::UpdateHeader::kResponseFormatError:
+        ++mResponseCounters.mFormatError;
+        break;
+    case Dns::UpdateHeader::kResponseNameExists:
+        ++mResponseCounters.mNameExists;
+        break;
+    case Dns::UpdateHeader::kResponseRefused:
+        ++mResponseCounters.mRefused;
+        break;
+    default:
+        OT_ASSERT(false);
```

Comment:
seems too restrict? 

```suggestion
        ++mResponseCounters.mOther;
```

Reply:
Every response code is obtained from `ErrorToDnsResponseCode` therefore I think there won't be any other response code types.

Reply:
I agree with @simonlingoogle. 
We may add other response codes in future and then having an assert in the stat/counter collection seems to be too strict.